### ISSUE TITLE
Add support to search for fields which are empty/not-empty

### ISSUE THREAD

**Issue by jwbensley** [k01ek/netbox-bgp#143]
**Is your feature request related to a problem? Please describe.**
It is not possible to search for custom fields which are empty/not-empty. For example, if I want to filter BGP sessions for all where a certain field is empty or is not empty, it's not possible.

**Describe the solution you'd like**
NetBox added support for an `__empty` filter to string fields, but only on native NetBox objects. There is now a request open to add it to custom fields, but for lists: https://github.com/netbox-community/netbox/issues/11538

Would it be possible to have this supported in this plugin? To search for empty/not-empty fields (both the fields native implemented by the plugin and custom fields)?

**Describe alternatives you've considered**
There is no way to work around this limition in the NetBox GUI or API right now. Instead you have to pull data from NetBox, then filter it in your own code, which results it's pulling lots of unneeded data from NetBox.

**Additional context**
None. Hopefully the request is clear.

**Comment by k01ek** [comment]
You can use NetBox Reports to check object fields .
For example:
`for session in BGPSession.objects.all():
      if session.cf['somefield'] is None:
         self.log_warning(session, f"somefield is empty!")`

**Comment by jwbensley** [comment]
Yeah that is a possibility, but what I was hoping for was not having to loop over every single BGP session in NetBox.

**Comment by candlerb** [comment]
I don't know if this helps: https://github.com/netbox-community/netbox/issues/11538#issuecomment-1635839720

**Comment by jwbensley** [comment]
This does kinda do what I want, so thanks @candlerb for the pointer, although I am not sure about using this in production.
As you point out in the NetBox issue, it is undocumented. Also that issue was closed without response so I'm leaning towards it being unsupported too? (even though this coded is referenced in the units tests as you point out). It's also not supported on any field type. I don't like having one solution for some fields, but not for others.

So for these reasons I will just work around this in code that receives the API response.
